lib/plug/session/cookie.ex

```diff
@@ -37,7 +37,17 @@ defmodule Plug.Session.COOKIE do
 
     * `:serializer` - cookie serializer module that defines `encode/1` and
       `decode/1` returning an `{:ok, value}` tuple. Defaults to
-      `:external_term_format`.
+      `:external_term_format`;
+
+    * `:active_secret_key_bases` - a list of secret key bases that are active
```

Comment:
Not a big fan of the word "active". Maybe "readonly" as you suggested? Fallback? Backup?

Reply:
I'm unsure what best describes. I used `active` as the secret is an active in use secret that is valid for a user. It is possible from one hosts perspective that an entry in this list is a future `:secret_key_base`. Today I am unsure which is clearer `fallback` or `readonly`.

Reply:
I think `accepted` could also work here.

The clarify of the name also depends on whether or not we expect the primary signing key to also appear in this list. I think it should, and maybe we should make that an invariant.

If we were starting from scratch, the most explicit names might be `signing_key` and `verification_keys` (or something along those lines).

Naming things. 😉

Reply:
We could take the approach of adding `signing_key`, `encryption_key`, `verification_keys`, decryption_keys` where they default to `:secret_key_base` or the base wrapped in list. These options would be in the cookie store, and not plug keys. That would be clearer and backwards compatible but add complexity. I do like that design as it allows the most explicit behavior. Concerned about having two ways of doing something.

CSRF could also have its own `signing_key` and `verification_keys` - note these don't need to be same as cookie store.